### PR TITLE
Correct comment for AtomicIncreaseSize method

### LINKED ISSUE #6856: Correct comment for AtomicIncreaseSize() method
Currently the comment says: 
```
	// AtomicIncreaseSize tries to increase the size of the node group atomically.
	// - If the method returns nil, it guarantees that delta instances will be added to the node group
	//   within its MaxNodeProvisionTime. The function should wait until node group size is updated.
	//   The cloud provider is responsible for tracking and ensuring successful scale up asynchronously.
	// - If the method returns an error, it guarantees that no new instances will be added to the node group
	//   as a result of this call. The cloud provider is responsible for ensuring that before returning from the method.
	// Implementation is optional. If implemented, CA will take advantage of the method while scaling up
	// GenericScaleUp ProvisioningClass, guaranteeing that all instances required for such a ProvisioningRequest
	// are provisioned atomically.
```
1)  GenericScaleUp should be replaced with BestEffortAtomicScaleUp

2) 	
```
    // - If the method returns nil, it guarantees that delta instances will be added to the node group
	//   within its MaxNodeProvisionTime. The function should wait until node group size is updated.
	//   The cloud provider is responsible for tracking and ensuring successful scale up asynchronously.
	// - If the method returns an error, it guarantees that no new instances will be added to the node group
	//   as a result of this call. The cloud provider is responsible for ensuring that before returning from the method.
```
this part is wrong. 
@kisieland what would be the right description?

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?

/kind documentation

Fixes #6856

### CODE REVIEW

**Comment by yaroslava-serdiuk** [comment]
cc: @aleksandra-malinowska @MaciekPytel

File: cluster-autoscaler/cloudprovider/cloud_provider.go [hunk 1]
```diff
@@ -183,14 +183,14 @@ type NodeGroup interface {
 	IncreaseSize(delta int) error
 
 	// AtomicIncreaseSize tries to increase the size of the node group atomically.
-	// - If the method returns nil, it guarantees that delta instances will be added to the node group
-	//   within its MaxNodeProvisionTime. The function should wait until node group size is updated.
-	//   The cloud provider is responsible for tracking and ensuring successful scale up asynchronously.
+	// - If the method returns nil, it guarantees that either delta instances will be added to the node group
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/cloudprovider/cloud_provider.go:186]: I'd remove the phrasing with 'guarantee'. 

We can't really have a guarantee that all nodes will register successfully until they do, and caller of this method should be able to handle this case later on (and cluster autoscaler does so, cleaning up the instances as usual).

Maybe sth like:

"AtomicIncreaseSize tries to increase the size of the node group atomically. It returns error if requesting the entire delta fails. The method doesn't wait until the new instances appear.

Implementation is optional. Implementation of this method generally requires external cloud provider support for atomically requesting multiple instances. If implemented, CA will < insert the rest of the description > "

?

Comment (yaroslava-serdiuk) [reply cluster-autoscaler/cloudprovider/cloud_provider.go]: SGTM, done

**Pull request merged**